### PR TITLE
Remove secret create restriction by resourcenames

### PR DESCRIPTION
**Pull request by francis-nijay**
# Description
Change to remove secret access restriction based on resourcenames. 

# GitHub Issues
| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1190 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?

- [x] Tested cluster onboard, assign and unassign license
- [x] Tested installation of csm through DN

### CODE REVIEW

**Review by abhi16394 — APPROVED** [review]
LGTM

File: operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml [hunk 1]
```diff
@@ -42,16 +42,16 @@ rules:
...
     verbs: ["list", "watch"]
   - apiGroups: [""]
     resources: ["persistentvolumes"]
     verbs: ["list", "watch"]
-  - apiGroups: ["storage.k8s.io"]
-    resources: ["storageclasses"]
-    verbs: ["list", "watch", "create", "delete"]
+  - apiGroups: [ "storage.k8s.io" ]
+    resources: [ "storageclasses" ]
+    verbs: [ "list", "watch", "create", "delete" ]
```
Comment (bpjain2004) [inline operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml:54]: please remove the redundant spaces here

File: operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml [hunk 2]
```diff
@@ -67,21 +67,12 @@ rules:
...
     verbs: ["list", "watch", "get"]
-  - apiGroups: [""]
-    resources: ["configmaps"]
-    verbs: ["get", "create", "delete", "update"]
-  - apiGroups: ["storage.dell.com"]
-    resources: ["containerstoragemodules"]
-    verbs: ["list", "create", "get", "delete", "watch"]
+  - apiGroups: [ "storage.dell.com" ]
+    resources: [ "containerstoragemodules" ]
+    verbs: [ "list", "get", "watch" ]
```
Comment (bpjain2004) [inline operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml:72]: please remove the redundant spaces here

**Comment by bpjain2004** [comment]
Please check the discovery service logs for any errors, once you onboard a cluster running the client with the above changes.

File: operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml (hunk 1 above)
Comment (francis-nijay) [reply operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml]: removed

File: operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml (hunk 2 above)
Comment (francis-nijay) [reply operatorconfig/clientconfig/apexconnectivityclient/v1.0.0/statefulset.yaml]: removed

**Review by bpjain2004 — APPROVED** [review]

**Review by jooseppi-luna — APPROVED** [review]
yes

**Pull request merged**